src/TestCase.ts

```diff
@@ -0,0 +1,241 @@
+import * as path from "path";
+import * as fs from "fs";
+import * as yaml from "js-yaml";
+import * as vscode from "vscode";
+import { Position, Range, Selection } from "vscode";
+import { Clipboard } from "./Clipboard";
+import NavigationMap from "./NavigationMap";
+import { ThatMark } from "./ThatMark";
+import {
+  ActionType,
+  PartialTarget,
+  PrimitiveTarget,
+  SelectionWithEditor,
+  Target,
+} from "./Types";
+
+export type SerializedPosition = {
+  line: number;
+  character: number;
+};
+
+export type SerializedRange = {
+  start: SerializedPosition;
+  end: SerializedPosition;
+};
+
+export type SerializedSelection = {
+  anchor: SerializedPosition;
+  active: SerializedPosition;
+};
+
+export function serializeRange(range: Range): SerializedRange {
+  return {
+    start: serializePosition(range.start),
+    end: serializePosition(range.end),
+  };
+}
+
+export function serializeSelection(selection: Selection): SerializedSelection {
+  return {
+    active: serializePosition(selection.active),
+    anchor: serializePosition(selection.anchor),
+  };
+}
+
+export function serializePosition(position: Position): SerializedPosition {
+  return { line: position.line, character: position.character };
+}
+
+type TestCaseCommand = {
+  actionName: ActionType;
+  partialTargets: PartialTarget[];
+  extraArgs: any[];
+};
+
+type TestCaseContext = {
+  talonCommand: string;
+  thatMark: ThatMark;
+  targets: Target[];
```

Comment:
It's a funny smell to have `targets` here but the actual cursorless command somewhere else

Reply:
Not sure I follow. Do you mean you would rather construct test case with a single argument/type ex. `{ command, talonCommand, thatMark, targets }` ?